accounts/abi/bind/backends/simulated_test.go

```diff
@@ -1114,3 +1114,41 @@ func TestSimulatedBackend_CallContractRevert(t *testing.T) {
 		sim.Commit()
 	}
 }
+
+func TestSimulatedBackend_Reorg(t *testing.T) {
```

Comment:
Can you add a short comment about the intention of the test?
Something like
```
TestSimulatedBackend_Reorg tests the reorg functionality.
Hereby 5 blocks are inserted into the chain and a reorg of 
the last 4 blocks to 6 different blocks occurs.
Resulting in a chain of 7 blocks.
```

Reply:
I describe the steps now:
```pre
// TestSimulatedBackend_ReorgTx tests the Reorg capability of the simulated
// backend. It runs these steps:
// 1. Setup `n` accounts and fund them. This is necessary since the account
//    nonce prevents transactions of the same account to from being re-ordered.
// 2. Deploy the `Counter` contract.
// 3. Send `n` transactions and confirm them.
// 4. Use a permutation to re-order the transactions in a simulated chain Reorg.
// 5. Emulate the `counter` variable of the contract and check that the
//    transactions are correctly confirmed or reverted.
```